subsys/bluetooth/audio/has_internal.h

```diff
@@ -8,9 +8,121 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-#define BT_HAS_FEAT_HEARING_AID_TYPE_MASK 0x03
+/* Control Point opcodes */
+#define BT_HAS_OP_READ_PRESET_REQ        0x01
+#define BT_HAS_OP_READ_PRESET_RSP        0x02
+#define BT_HAS_OP_PRESET_CHANGED         0x03
+#define BT_HAS_OP_WRITE_PRESET_NAME      0x04
+#define BT_HAS_OP_SET_ACTIVE_PRESET      0x05
+#define BT_HAS_OP_SET_NEXT_PRESET        0x06
+#define BT_HAS_OP_SET_PREV_PRESET        0x07
+#define BT_HAS_OP_SET_ACTIVE_PRESET_SYNC 0x08
+#define BT_HAS_OP_SET_NEXT_PRESET_SYNC   0x09
+#define BT_HAS_OP_SET_PREV_PRESET_SYNC   0x0a
+
+/* Application error codes */
+#define BT_HAS_ERR_INVALID_OPCODE         0x80
+#define BT_HAS_ERR_WRITE_NAME_NOT_ALLOWED 0x81
+#define BT_HAS_ERR_PRESET_SYNC_NOT_SUPP   0x82
+#define BT_HAS_ERR_OPERATION_NOT_POSSIBLE 0x83
+#define BT_HAS_ERR_INVALID_PARAM_LEN      0x84
```

Comment:
Not exactly, the error codes can be handled by the stack internally:

- Errors like BT_HAS_ERR_INVALID_OPCODE and BT_HAS_ERR_PRESET_SYNC_NOT_SUPP depend on Kconfig settings.
- BT_HAS_ERR_INVALID_PARAM_LEN: The PDU length can be verified as well by request handler;
- BT_HAS_ERR_WRITE_NAME_NOT_ALLOWED: Depends on BT_HAS_PRESET_NAME_DYNAMIC plus the Preset property

If the Client tries to change the preset but the operation cannot be performed at the time BT_HAS_ERR_OPERATION_NOT_POSSIBLE will be sent. But the API can use one of the errno ecodes instead.

This leads to conclusion that those error codes can be internal.

Reply:
> This leads to conclusion that those error codes can be internal.

Possibly yes. It will depend on how the client API ends up being. For example the GATT API will generally return the ATT error codes on read/write callbacks. 

I am not sure we want to transform the HAS error codes to errno error codes (also not sure that we don't want to do that...), as that will result in a client API that will use errno error codes in the callback, rather than ATT/HAS error codes. 

We can postpone this discussion until we have the client PR open :)

Reply:
OK, so I'll leave the error codes defines as it is for now